added docker files for local test environment

wolfgangwalther opened the pull request:
This adds a `test/Dockerfile` and `test/docker-compose.yml` to spin up a local dev environment quickly.

theory commented on .gitignore:
```diff
@@ -30,3 +30,6 @@ test/build
 .DS_Store
+
+# symlinked docker run script
+/run
```
Is this something on your own box, or does it get generated somehow?

wolfgangwalther replied:
This is the file `test/docker_make_test.sh` which is mapped as a volume in `docker-compose.yml` to `/pgtap/run` inside the container. And because the main directory is mapped as `/pgtap` as well, the file is actually created on the host.

The mapping is just there to be able to run `docker-compose exec test run` instead of `docker-compose exec test test/docker_make_test.sh`.

theory replied:
I suggest copying it to some other location; no need to pollute the repo on every run. Does it need to go there? If so, a better more temporary-looking name might be useful.

wolfgangwalther replied:
Agreed. The main point why I did it this way, was because I didn't want to put the script in the main directory itself - because it's only needed for running the tests. Having it still show up there, kind of defeats that... ;)

I changed it a bit, I think the way is better now.

wolfgangwalther commented:
rebased on current master; squashed; updated base image from pg 12.3 to pg 12.4

wolfgangwalther commented:
Updated base image from pg 12.4 to pg 13.0.

Added a tmpfs volume for the postgres data directory to allow switching the pg version easier by just changing the number in the dockerfile. Before that the existing volume had to be removed manually, because the data directory was not compatible between versions.
